### PR TITLE
Add nested plugin configuration completion for Maven Plugins

### LINKED ISSUE #4: Content assist & hover for complex <configuration> element
From https://github.com/AObuchow/lsp4xml-extensions-maven/issues/85

m2e has it.
For some plugin configuration elements (like maven-dependency-plugin/copy/artifactItems), m2e is able to complete the children of the element.
We should copy the responsible m2e code into the Lemminx extension

Difficulty: **..
Value (frequency*time saved): ****

### PR DESCRIPTION
**Pull request by AObuchow**
Fix #4

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

### CODE REVIEW

File: lemminx-maven/src/test/java/org/eclipse/lemminx/maven/test/LocalPluginTest.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public void testCompleteConfigurationParametersDuplicates() throws IOException,
 		List<CompletionItem> completions = languageService.doComplete(createDOMDocument("/pom-duplicate-configuration-completion.xml", languageService),
 				new Position(15, 4), new SharedSettings())
 			.getItems();
-		assertTrue(completions.stream().map(CompletionItem::getLabel).filter("compilePath"::equals).count() == 1);
+		assertTrue(completions.stream().map(CompletionItem::getLabel).filter("annotationProcessors"::equals).count() == 1);
```
Comment (AObuchow) [inline lemminx-maven/src/test/java/org/eclipse/lemminx/maven/test/LocalPluginTest.java:111]: compilePath is not actually an editable parameter for maven-compiler-plugin, so this test was faulty

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java [hunk 2]
```diff
@@ -123,14 +123,60 @@ public void onTagOpen(ICompletionRequest request, ICompletionResponse response)
...
-					.map(parameter -> toTag(parameter.getName(), MavenPluginUtils.getMarkupDescription(parameter, supportsMarkdown), request))
-					.forEach(response::addCompletionItem);
+		Set<MojoParameter> parameters;
+		try {
+			parameters = MavenPluginUtils.collectPluginConfigurationMojoParameters(request, cache, repoSession, pluginManager, buildPluginManager, mavenSession);
+		} catch (PluginResolutionException | PluginDescriptorParsingException | InvalidPluginDescriptorException e) {
+			e.printStackTrace();
+			return null;
 		}
+		//DOMNode node = request.getNode();
```
Comment (AObuchow) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java:143]: Remove

**Comment by AObuchow** [comment]
This works although maybe it could be factored further with MavenHoverParticipant.

image

I had to make modifications to the sort text of maven properties as `$` seems to take priority over alpha characters.
It also seems that lemminx's snippets (for comments) take priority at the top of the completion list. I'm not sure if this can be avoided downstream.

**Comment by mickaelistria** [comment]
That looks good to me. Feel free to merge.

**Pull request merged**